### PR TITLE
Comboboxes for selectors

### PR DESCRIPTION
**Pull request by tylermercer**

### CODE REVIEW

**Comment by tylermercer** [comment]
For some reason just touching (i.e. focusing and then blurring) the autocomplete field doesn't trigger the "Required" warning.... I don't know why that is.

File: src/components/formik/Combobox.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Field, useField } from "formik";
+import { TextField } from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+const AutocompleteTextField = ({ className, name, label, options }) => {
+  const [field, meta, helpers] = useField(name);
+  const error = !!meta.touched && !!meta.error;
+  const helperText = error ? meta.error : " ";
+
+  console.log({ error });
```
Comment (bradenwatkins) [inline src/components/formik/Combobox.js]: Can you remove this console.log?

File: src/components/formik/Combobox.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { Field, useField } from "formik";
+import { TextField } from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+const AutocompleteTextField = ({ className, name, label, options }) => {
```
Comment (bradenwatkins) [inline src/components/formik/Combobox.js]: Can you rename this to "Combobox"?

File: src/pages/EmailFlow/AddOffer/BonusesSubForm.js [hunk 3]
```diff
@@ -4,11 +4,11 @@ import { FieldArray } from "formik";
 import { makeStyles } from "@material-ui/core/styles";
 import IconButton from "@material-ui/core/IconButton";
 import Typography from "@material-ui/core/Typography";
-import MenuItem from "@material-ui/core/MenuItem";
 import DeleteIcon from "@material-ui/icons/Delete";
 import AddIcon from "@material-ui/icons/Add";
 import * as cx from "classnames";
 
+import AutocompleteTextField from "../../../components/formik/AutocompleteTextField";
```
Comment (bradenwatkins) [inline src/pages/EmailFlow/AddOffer/BonusesSubForm.js]: What is the difference between Autocomplete and combobox?

File: src/components/formik/Combobox.js (hunk 2 above)
Comment (tylermercer) [reply src/components/formik/Combobox.js]: Good catch

File: src/components/formik/Combobox.js (hunk 1 above)
Comment (tylermercer) [reply src/components/formik/Combobox.js]: Ah yep, good catch

File: src/pages/EmailFlow/AddOffer/BonusesSubForm.js (hunk 3 above)
Comment (tylermercer) [reply src/pages/EmailFlow/AddOffer/BonusesSubForm.js]: The combobox is essentially an html `select`. It has a specific set of choices. The autocomplete is just a free-response text field that offers suggestions. Both use the Autocomplete MUI component under the hood, but in different ways. Getting both functionalities in one component proved difficult with Formik.

**Comment by tylermercer** [comment]
Realized I need to actually use the Combobox component; right now it's only used for one of the Selects instead of for all of them.

**Pull request merged**